### PR TITLE
Home frontend news

### PR DESCRIPTION
**Pull request by ghost**
### Summary
Now, in news tab, you can see a list of news and a add button. Pull down shows loading icon.

### Card
https://trello.com/c/zrzQ076v/35-u4-home-page-frontend-news

### Screen Recordings
news

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt [hunk 1]
```diff
@@ -1,13 +1,41 @@
...
+            // TODO FOR NEXT PR (Backend): change this list for data from a request to DB/API
+            adapter = HomeNewsViewAdapter(listOf(News("Detuvieron a Alberto Samid en Belice",
+                    "En Belice, fue detenido por Interpol Alberto Samid. Nuestra" +
+                            " política es clara y transparente. Quienes tienen deudas con la" +
+                            " justicia, deben"), News("Susto para Lionel Scaloni en España:" +
+                    " fue atropellado por un auto y sufrió lesiones", "El DT de la " +
+                    "Argentina iba en bicicleta y fue embestido en el estacionamiento del " +
+                    "colegio de sus hijas. Fue atendido en un hospital y ya recibió el alta.")))
+            layoutManager = LinearLayoutManager(context)
+            addItemDecoration(DividerItemDecoration(context, DividerItemDecoration.VERTICAL))
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt:27]: What is the purpose of this line?

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsViewAdapter.kt [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+import android.support.v7.widget.RecyclerView
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.TextView
+import ar.com.wolox.android.R
+import ar.com.wolox.android.example.model.News
+
+class HomeNewsViewAdapter(private val dataSet: List<News>) :
```
Comment (lmiotti) [inline app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsViewAdapter.kt:9]: The idea of this is perfect. As a matter of guidelines we are migrating to ListAdapter, try to extend this adapter to ListAdapter class

File: app/src/main/res/layout/fragment_home_news.xml [hunk 3]
```diff
@@ -1,7 +1,41 @@
 <?xml version="1.0" encoding="utf-8"?>
-<android.support.constraint.ConstraintLayout
-    xmlns:android="http://schemas.android.com/apk/res/android" android:layout_width="match_parent"
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home_news.xml]: Move xlmns to a new line

File: app/src/main/res/layout/fragment_home_news_item.xml [hunk 4]
```diff
@@ -0,0 +1,74 @@
+<?xml version="1.0" encoding="utf-8"?>
+<android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home_news_item.xml]: Same as above

File: app/src/main/res/layout/fragment_home_news.xml [hunk 5]
```diff
@@ -1,7 +1,41 @@
...
+    android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:background="#F00070"> <!-- TODO: hardcoded because is for testing -->
+    xmlns:app="http://schemas.android.com/apk/res-auto" >
```
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home_news.xml]: xmlns have to go all together

File: app/src/main/res/layout/fragment_home_news.xml [hunk 6]
```diff
@@ -1,7 +1,41 @@
...
+
+    </android.support.v4.widget.SwipeRefreshLayout>
+
+    <android.support.design.widget.FloatingActionButton
+        android:id="@+id/vHomeNewsAddButton"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:layout_marginBottom="@dimen/spacing_large"
+        android:layout_marginEnd="@dimen/spacing_large"
+        android:src="@drawable/baseline_add_white_18dp"
```
Comment (lmiotti) [inline app/src/main/res/layout/fragment_home_news.xml]: Can we change this drawable name/id?

**Review by lmiotti — CHANGES_REQUESTED** [review]
It's quite good, but we need to do some corrections

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt (hunk 1 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsFragment.kt:27]: If I don't add this line, the list does not have separators between elements. This adds the thin light grey lines between elements.

File: app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsViewAdapter.kt (hunk 2 above)
Comment (ghost) [reply app/src/main/java/ar/com/wolox/android/example/ui/example/home/HomeNewsViewAdapter.kt:9]: ✔ **Done**

File: app/src/main/res/layout/fragment_home_news.xml (hunk 3 above)
Comment (ghost) [reply app/src/main/res/layout/fragment_home_news.xml]: ✔ **Done**

File: app/src/main/res/layout/fragment_home_news_item.xml (hunk 4 above)
Comment (ghost) [reply app/src/main/res/layout/fragment_home_news_item.xml]: ✔ **Done**

File: app/src/main/res/layout/fragment_home_news.xml (hunk 5 above)
Comment (ghost) [reply app/src/main/res/layout/fragment_home_news.xml]: ✔ **Done**

File: app/src/main/res/layout/fragment_home_news.xml (hunk 6 above)
Comment (ghost) [reply app/src/main/res/layout/fragment_home_news.xml]: ✔ **Changed** to ```ic_plus_white```

**Review by lmiotti — APPROVED** [review]
tenor

**Pull request merged**